rclcpp/include/rclcpp/clock.hpp

```diff
@@ -121,6 +123,17 @@ class Clock
     JumpHandler::post_callback_t post_callback,
     const rcl_jump_threshold_t & threshold);
 
+  /**
+   * Get current time from the time source specified by Clock::get_clock_type() const noexcept.
+   *
+   * \param now. A pointer to time point value, which will be
+   * set to the current time according to the underlying clock.
+   * \return RCUTILS_RET_OK if time was successfully retrieved
+   */
+  RCLCPP_PUBLIC
+  rcutils_ret_t
+  get_now_as_time_point_for_logging(rcutils_time_point_value_t * now);
```

Comment:
Why is `_for_logging` in the name? If it's to differentiate from returning a C++ time point type, I'd use `get_now_as_rcutils_time_point()` instead. Putting the current user of the function in the name doesn't make much sense to me.

Reply:
On this point, why not use the existing `now()` method on this class and then call this:

https://github.com/ros2/rclcpp/blob/b8f723708757b0d0b7f29de47544798bafc24a6e/rclcpp/include/rclcpp/time.hpp#L101-L103

It returns an `rcl_time_point_value_t`, but that's the same thing as the rcutils one:

https://github.com/ros2/rcl/blob/e6739884151c302838cfbe908fc9963ef4836f8f/rcl/include/rcl/time.h#L44

That way you don't even need a new method on this class. Right?

Reply:
(answering my own question, perhaps) Maybe because it could throw? But then you could try-catch in the RCLCPP macro.

Reply:
> Why is _for_logging in the name?

The name came from Dirk's comment
https://github.com/ros2/rclcpp/pull/879#discussion_r333060831.

If we keep the idea of reusing rcutils macro, I can't use the `now` method along with `nanoseconds` since the [rcutils macro](https://github.com/ros2/rcutils/blob/0815d74809bae0158b8e92ca502017eec5779fa4/resource/logging_macros.h.em#L181) takes a pointer as parameter. The compiler won't like that.

Reply:
> ... we could mangle the name to something like: `get_now_as_time_point_for_logging`.

I am fine with anything - that a merely a suggestion. Naming it `get_now_as_rcutils_time_point` as William mentioned above is fine with me too.

Reply:
> If we keep the idea of reusing rcutils macro, I can't use the now method along with nanoseconds since the rcutils macro takes a pointer as parameter. The compiler won't like that.

Wait what? Can you not:

```c++
auto rcutils_time_point = clock->now().nanoseconds();
MACRO(&rcutils_time_point);
```

Maybe I don't understand what you mean?

Reply:
> Wait what? Can you not:

No. My PR for [rcutils](https://github.com/ros2/rcutils/pull/183) makes use of a function rather than a plain value. These two PRS are connected.

Reply:
Well then this:

```c++
std::function<rcutils_ret_t (rcutils_time_point_value_t *)> get_time_point = [&clock](rcutils_time_point_value_t * time_point) -> rcutils_ret_t {
  *time_point = clock->now().nanoseconds();
}
```

I just don't think we want/need another way to get an `rcl_time_point_value_t`/`rcutils_time_point_value_t` from the Clock/Time objects.